src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,26 +1,1711 @@
+            engine_fire_push_buttons.is_released(self.backup_channel_engine_number),
+            overhead_panel.apu_bleed_is_on(),
+            apu_bleed_valve_is_open,
+            overhead_panel.cross_bleed_mode(),
+            cross_bleed_valve_is_open,
+        );
+    }
+}
+
+struct BleedMonitoringComputerChannel {
+    engine_number: usize,
+    ip_compressor_pressure: Pressure,
+    hp_compressor_pressure: Pressure,
+    // Pressure between IP/HP valves and the PRV
+    transfer_pressure: Pressure,
+    // Pressure after PRV
+    regulated_pressure: Pressure,
+    prv_open_amount: Ratio,
+    hpv_open_position: Ratio,
+    esv_is_open: bool,
+    is_engine_bleed_pushbutton_auto: bool,
+    is_engine_fire_pushbutton_released: bool,
+    is_apu_bleed_valve_open: bool,
+    is_apu_bleed_on: bool,
+    hpv_pid: Pid<f64>,
+    hpv_output: f64,
+    prv_pid: Pid<f64>,
+    prv_output: f64,
+    cross_bleed_valve_selector: CrossBleedValveSelectorMode,
+    cross_bleed_valve_is_open: bool,
+}
+impl BleedMonitoringComputerChannel {
+    fn new(engine_number: usize) -> Self {
+        Self {
+            engine_number,
+            ip_compressor_pressure: Pressure::new::<psi>(0.),
+            hp_compressor_pressure: Pressure::new::<psi>(0.),
+            transfer_pressure: Pressure::new::<psi>(0.),
+            regulated_pressure: Pressure::new::<psi>(0.),
+            hpv_open_position: Ratio::new::<percent>(0.),
+            prv_open_amount: Ratio::new::<percent>(0.),
+            esv_is_open: false,
+            is_engine_bleed_pushbutton_auto: true,
+            is_engine_fire_pushbutton_released: false,
+            is_apu_bleed_valve_open: false,
+            is_apu_bleed_on: false,
+            hpv_pid: Pid::new(0.05, 0., 0., 1., 1., 1., 1., 65.),
+            hpv_output: 0.,
+            prv_pid: Pid::new(0., 0.01, 0., 1., 1., 1., 1., 46.),
+            prv_output: 0.,
+            cross_bleed_valve_selector: CrossBleedValveSelectorMode::Auto,
+            cross_bleed_valve_is_open: false,
+        }
+    }
+
+    fn update(
+        &mut self,
+        sensors: &impl EngineBleedDataProvider,
+        is_engine_bleed_pushbutton_auto: bool,
+        is_engine_fire_pushbutton_released: bool,
+        is_apu_bleed_on: bool,
+        apu_bleed_valve_is_open: bool,
+        cross_bleed_valve_selector: CrossBleedValveSelectorMode,
+        cross_bleed_valve_is_open: bool,
+    ) {
+        self.ip_compressor_pressure = sensors.ip_pressure();
+        self.hp_compressor_pressure = sensors.hp_pressure();
+        self.transfer_pressure = sensors.transfer_pressure();
+        self.regulated_pressure = sensors.regulated_pressure();
+
+        self.hpv_output = self
+            .hpv_pid
+            .next_control_output(self.transfer_pressure.get::<psi>())
+            .output;
+        self.prv_output = self
+            .prv_pid
+            .next_control_output(self.regulated_pressure.get::<psi>())
+            .output;
```

Comment:
I just noticed those are not using a time dt to compute the control, so it would probably do strange stuff when dt is changing from frame to frame. This could be enough but take care :D

Reply:
Oh, this is a good point. The PID library I'm using even explicitly states this as an assumption :D I will have to look into it more. Thanks for pointing this out!

Reply:
I was thinking of using this lib to do some quick tests and noticed it requires fixed steps indeed. Could be worth implementing your own pid, or maybe I can try to share the one I use for autobrake, which is in a recursive form and doesn't require a dt value